docs/lang-go.md

```diff
@@ -35,10 +35,11 @@ import (
 )
 
 func main() {
-    store := wasmtime.NewStore(wasmtime.NewEngine())
-    module, err := wasmtime.NewModuleFromFile(store, "gcd.wat")
+    engine := wasmtime.NewEngine()
+    module, err := wasmtime.NewModuleFromFile(engine, "gcd.wat")
     check(err)
-    instance, err := wasmtime.NewInstance(module, []*wasmtime.Extern{})
+    store := wasmtime.NewStore(engine)
```

Comment:
I think it would make sense to move this just after `engine := wasmtime.NewEngine()`.

Reply:
Although I have moved this line, I can not reason about, can you share your point?

Reply:
I can't really. It is more of a "it feels better" than "I have a concrete reason it is better".